src/main/java/com/cburch/logisim/std/wiring/PinAttributes.java

```diff
@@ -41,27 +42,36 @@ public boolean isHidden() {
   public static final Attribute<String> ATTR_DUMMY = new DummyAttr("type");
   public static PinAttributes instance = new PinAttributes();
 
-  private static final List<Attribute<?>> ATTRIBUTES =
-      Arrays.asList(
-          StdAttr.FACING,
-          Pin.ATTR_TYPE,
-          StdAttr.WIDTH,
-          Pin.ATTR_TRISTATE,
-          Pin.ATTR_PULL,
-          StdAttr.LABEL,
-          StdAttr.LABEL_FONT,
-          RadixOption.ATTRIBUTE,
-          PROBEAPPEARANCE,
-          ATTR_DUMMY);
+  private List<Attribute<?>> myAttributes;
 
   BitWidth width = BitWidth.ONE;
   boolean threeState = false; // true;
   int type = EndData.INPUT_ONLY;
   Object pull = Pin.PULL_NONE;
+  Long initialValue = 0L;
+
+  public PinAttributes() {
+    updateAttributes();
+  }
+
+  private void updateAttributes() {
+    myAttributes = new ArrayList<Attribute<?>>();
+    myAttributes.add(StdAttr.FACING);
+    myAttributes.add(Pin.ATTR_TYPE);
+    myAttributes.add(StdAttr.WIDTH);
+    myAttributes.add(Pin.ATTR_TRISTATE);
+    myAttributes.add(Pin.ATTR_PULL);
+    myAttributes.add(StdAttr.LABEL);
+    myAttributes.add(StdAttr.LABEL_FONT);
+    myAttributes.add(RadixOption.ATTRIBUTE);
+    if ((type == EndData.INPUT_ONLY) && !threeState) myAttributes.add(Pin.ATTR_INITIAL);
```

Comment:
Can we have this line as 

```java
if ((type == EndData.INPUT_ONLY) && !threeState) {
    myAttributes.add(Pin.ATTR_INITIAL);
}
```

for better readability?

Reply:
done